packages/vm/src/testing/mock.rs

```diff
@@ -33,130 +38,136 @@ pub fn mock_backend_with_balances(
     }
 }
 
-/// Length of canonical addresses created with this API. Contracts should not make any assumptions
-/// what this value is.
-///
-/// The value here must be restorable with `SHUFFLES_ENCODE` + `SHUFFLES_DECODE` in-shuffles.
-/// See <https://oeis.org/A002326/list> for a table of those values.
-const CANONICAL_LENGTH: usize = 64; // n = 32
-
-const SHUFFLES_ENCODE: usize = 10;
-const SHUFFLES_DECODE: usize = 2;
-
 /// Zero-pads all human addresses to make them fit the canonical_length and
 /// trims off zeros for the reverse operation.
 /// This is not really smart, but allows us to see a difference (and consistent length for canonical adddresses).
 #[derive(Copy, Clone)]
-pub struct MockApi {
-    /// Length of canonical addresses created with this API. Contracts should not make any assumptions
-    /// what this value is.
-    canonical_length: usize,
-    /// When set, all calls to the API fail with BackendError::Unknown containing this message
-    backend_error: Option<&'static str>,
+pub enum MockApi {
```

Comment:
Can we keep this a struct and move the enum in a field? The API might need other settings that just the address settings in the future.

Reply:
I don't really know what field we would possibly want to add to both variants, but makes sense to keep the enum private in any case